packages/discord.js/src/structures/Base.ts

```diff
@@ -0,0 +1,8 @@
+import { Client } from './client/Client';
+
+export class Base {
+	public readonly client: Client;
+	public constructor(client: Client) {
+		this.client = client;
+	}
```

Comment:
```suggestion	
	public constructor(public readonly client: Client) {}
```

Reply:
Problem with inline properties is that they can't be documented, and it confuses developers who are pretty new to the TypeScript language.

Reply:
You can document them using some kinda ugly TSDoc: 

![using `@param`](https://user-images.githubusercontent.com/7608555/115933180-31312300-a443-11eb-8ab9-dd0d4def2aea.png)
or
![using inline TSDoc comment](https://user-images.githubusercontent.com/7608555/115933226-473ee380-a443-11eb-9a46-b887b00bdeab.png)

Reply:
> Problem with inline properties is that they can't be documented, and it confuses developers who are pretty new to the TypeScript language.

In addition to what Jonah said, I don't think we should worry about confusion on the lib end of things. You'd expect someone who's going to contribute to the lib to have knowledge of or at least be able to look up the concept of inline property declaration.

Reply:
> > Problem with inline properties is that they can't be documented, and it confuses developers who are pretty new to the TypeScript language.
> 
> In addition to what Jonah said, I don't think we should worry about confusion on the lib end of things. You'd expect someone who's going to contribute to the lib to have knowledge of or at least be able to look up the concept of inline property declaration.

That's a bold statement, not everyone learns at the same pace nor the same things. I for example, learnt about this feature after my third year of writing Node.js projects exclusively in TypeScript, after reading the source code from other people.

That being said, keeping the fields separated re-uses an already existing feature many users are comfortable with as the syntax is already present in plain JavaScript, likewise for users who come from other languages (C++, C#, Java...).

IMHO inline properties are a niche feature that serves no real purpose but to provide shorter syntax, there's no benefit on them and it confuses users.

Reply:
@kyranet agreed.